framework/src/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -613,6 +613,17 @@ public Store getStoreBefore(Tree tree) {
             // be the best possible.
             return null;
         }
+        return getStoreBefore(node);
+    }
+
+    /**
+     * @return The store immediately before a given {@link Node}.
+     */
+    public Store getStoreBefore(Node node) {
+        if (analyses == null || analyses.isEmpty()) {
```

Comment:
Isn't analyses guaranteed to be non-null? The field is final and set in the constructor. Can you see what happens when you remove the few null checks that exist?